sagemaker_processing/feature_transformation_with_sagemaker_processing_Dask/preprocess.py

```diff
@@ -0,0 +1,166 @@
+from __future__ import print_function
+from __future__ import unicode_literals
+import s3fs
+import dask.dataframe as dd
+import boto3
+import sys
+import time
+import json
+import time
+from dask.distributed import Client
+import argparse
+import os
+import warnings
+from tornado import gen
+
+import pandas as pd
+import numpy as np
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import StandardScaler, OneHotEncoder, LabelBinarizer, KBinsDiscretizer
+from sklearn.preprocessing import PolynomialFeatures
+from sklearn.compose import make_column_transformer
+
+from sklearn.exceptions import DataConversionWarning
+warnings.filterwarnings(action='ignore', category=DataConversionWarning)
+import joblib
+import dask.dataframe as dd
+s3_client = boto3.resource('s3')
+
+
+
+def get_resource_config():
+    resource_config_path = '/opt/ml/config/resourceconfig.json'
+    with open(resource_config_path, 'r') as f:
+        return json.load(f)
+
+def get_ip_from_host(host_name):
+    IP_WAIT_TIME = 300
+    counter = 0
+    ip = ''
+
+    while counter < IP_WAIT_TIME and ip == '':
+        try:
+            ip = socket.gethostbyname(host_name)
+            break
+        except:
+            counter += 1
+            time.sleep(1)
+
+    if counter == IP_WAIT_TIME and ip == '':
+        raise Exception("Exceeded max wait time of 300s for hostname resolution")
+
+    return ip
+
+
+
+def upload_objects(bucket, prefix, local_path):
+    try:
+        bucket_name = bucket #s3 bucket name
+        root_path = local_path # local folder for upload
+
+        s3_bucket = s3_client.Bucket(bucket_name)
+
+        for path, subdirs, files in os.walk(root_path):
+            for file in files:
+                s3_bucket.upload_file(os.path.join(path, file), prefix +'/output/'+file)
+
+    except Exception as err:
+        print(err)
```

Comment:
suggest to use log.exception from python logging instead of print

Reply:
Done